### PR TITLE
feat:  dcmaw-8544 user can see email on home page

### PR DESCRIPTION
**Pull request by jdubey**
# DCMAW-8544 | User can see their email on the home page

After successful login, decode the idToken to extract the e-mail, and display on the home page

Added additional functionality to JWTVerifier to decode the idToken without performing verification, since at the point of decoding, the idToken will already have been verified, whether it has been freshly update from BE or retrieved from secure store (as we wouldn't store an unverified token).  This is to prevent the need from having to make another JWKS call every time we decode.

Also refactored so that the initial biometrics/passcode request is for the idToken, not the access token.   The access token is retrieved from the secure store when the Developer Menu is visited, as that is when it is actually required.  This will invoke another biometric auth request, as that is the nature of how secure store works, i.e., every access to the store requires retrieval of the private key from the secure enclave, and every secure enclave access requires biometric authentication.

# Checklist

## Before raising your pull request:
- [x] Commit messages that conform to conventional commit messages
- [x] Ran the app locally ensuring it builds 
- [x] Ran the tests locally ensuring they pass on Build
- [x] Pull request has a clear title with a short description about the feature or update
~~- [ ] Created a `draft` pull request if it is not yet ready for review~~

## Before your pull request can be reviewed:
- [x] Met all of the acceptance criteria specified in the user story on Jira
- [x] Reviewed your own code to ensure you are following the style guidelines
- [x] Ran the app and tested the feature on a range of device sizes
      Please include iPod Touch, iPhone SE and iPhone 11 as a minimum.
- [x] Written Unit and Integration tests if needed

~~- [ ] Met all accessibility requirements?~~
    ~~- [ ] Checked dynamic type sizes are applied~~
    ~~- [ ] Checked VoiceOver can navigate your new code~~
    ~~- [ ] Checked a user can navigate only using a keyboard around your new code~~

## Before merging your pull request:
- [x] Ensure that the code coverage and SonarCloud checks have passed
- [x] Actioned and resolved all comments, reaching out to reviewers for clarifications if necessary.
- [x] Ran the app to ensure that no regressions have been caused by changes during code review.

### CODE REVIEW

**Review by mcgarj — APPROVED** [review]

File: Sources/Application/MainCoordinator.swift [hunk 1]
```diff
@@ -45,12 +46,18 @@ final class MainCoordinator: NSObject,
...
+                            throw SecureStoreError.unableToRetrieveFromUserDefaults
+                        }
+                        tokenHolder.idTokenPayload = try tokenVerifier.extractPayload(idToken)
                         updateToken()
                         action()
                     } catch {
-                        print("Error getting token: \(error)")
+                        loginCoordinator?.root.dismiss(animated: false)
+                        tokenHolder.accessToken = nil
+                        userStore.defaultsStore.set(nil, forKey: .accessTokenExpiry)
```
Comment (DylanRJ) [inline Sources/Application/MainCoordinator.swift:58]: Would calling `UserStorable.clearTokenInfo()` make sense here?

File: Sources/Login/LoginCoordinator.swift [hunk 2]
```diff
@@ -92,7 +95,16 @@ final class LoginCoordinator: NSObject,
                     root.pushViewController(networkErrorScreen, animated: true)
                 }
             }
+
         root.setViewControllers([rootViewController], animated: true)
+        if let startupError {
```
Comment (DylanRJ) [inline Sources/Login/LoginCoordinator.swift:100]: Wondering if we could give this a more descriptive name just so it's clear they user has been unable to either get the id token or verify it?

**Review by DylanRJ — APPROVED** [review]
There's a fair amount of complexity with the id token reading, verification and extraction. Might be useful to do an internal team recce at some point to make sure we're aligned 👍

File: Sources/Login/LoginCoordinator.swift [hunk 3]
```diff
@@ -49,27 +53,26 @@ final class LoginCoordinator: NSObject,
...
+                        throw SecureStoreError.unableToRetrieveFromUserDefaults
+                    }
+                    tokenHolder.idTokenPayload = try tokenVerifier.extractPayload(idToken)
                     windowManager.hideUnlockWindow()
                     root.dismiss(animated: false)
                     finish()
-                } catch SecureStoreError.unableToRetrieveFromUserDefaults,
-                        SecureStoreError.cantInitialiseData,
-                        SecureStoreError.cantRetrieveKey {
+                } catch {
```
Comment (DylanRJ) [inline Sources/Login/LoginCoordinator.swift:72]: Would we not want to add the error to the `startupError` property here?

**Pull request merged**